### PR TITLE
TNT-41403 Enable AEP mode via edgeConfigId SDK setting

### PR DESCRIPTION
**Pull request by XDex**
## Description

TNT-41403 Enable AEP mode via edgeConfigId SDK setting

## Related Issue

https://jira.corp.adobe.com/browse/TNT-41403

### CODE REVIEW

**Review by jasonwaters — COMMENTED** [review]
I noticed you made `aepEdgeService.js`.  But we don't really refer to anything else as a "service" in the node js sdk up to now.  For example, the `target.js` file has all the `executeDelivery` functionality within to handle delivery api request/responses.  For consistency can we rename `aepEdgeService.js` to just `aep.js` or `aepEdge.js` ?

Also, I think creating the abstraction at the level you have will work fine (having target.js do delivery api and aepEdge.js do aep).  But I wonder if it might be better to separate it the way it was done using the `createDeliveryApiMethod` function (see helper.js line 597).  Currently it conditionally returns an object with an `execute` method on it.  Up till now, the SDK returned either an object that goes to Delivery API or one that goes to decisioning engine.  But we can now add a condition to return an `aep execution object` that is responsible for translating requests/responses to the aep edge api.  What do you think?

**Comment by XDex** [comment]
> I noticed you made `aepEdgeService.js`. But we don't really refer to anything else as a "service" in the node js sdk up to now. For example, the `target.js` file has all the `executeDelivery` functionality within to handle delivery api request/responses. For consistency can we rename `aepEdgeService.js` to just `aep.js` or `aepEdge.js` ?

I wanted to align it with the SDK architecture doc, but sure, let me shorten it to `aepEdge.js`

> Also, I think creating the abstraction at the level you have will work fine (having target.js do delivery api and aepEdge.js do aep). But I wonder if it might be better to separate it the way it was done using the `createDeliveryApiMethod` function (see helper.js line 597). Currently it conditionally returns an object with an `execute` method on it. Up till now, the SDK returned either an object that goes to Delivery API or one that goes to decisioning engine. But we can now add a condition to return an `aep execution object` that is responsible for translating requests/responses to the aep edge api. What do you think?

Right, I considered this option, but it's a bit more restrictive, as we end up with somewhat redundant code paths in the AEP flow, like the HTTP client configuration performed in `target.js` (setting server host and request headers for Delivery API), which we'll still need to override downstream. So, I opted for a cleaner separation between Delivery and AEP flows..

**Comment by jasonwaters** [comment]
> Right, I considered this option, but it's a bit more restrictive, as we end up with somewhat redundant code paths in the AEP flow, like the HTTP client configuration performed in `target.js` (setting server host and request headers for Delivery API), which we'll still need to override downstream. So, I opted for a cleaner separation between Delivery and AEP flows..

Fair enough, I think the way you have it will work well.  If we see a reason to refactor to a lower level we can always do it later.

**Review by jasonwaters — APPROVED** [review]

File: packages/target-nodejs-sdk/test/index.spec.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ require("jest-fetch-mock").enableMocks();
 const TargetTools = require("@adobe/target-tools");
 const Visitor = require("@adobe-mcid/visitor-js-server");
 const target = require("../src/target");
+const aepEdgeService = require("../src/aepEdge");
```
[the remark this answers is not in the document]
Comment (XDex) [reply packages/target-nodejs-sdk/test/index.spec.js:17]: Thanks, missed those with the batch rename..

**Pull request merged**